test/es6/forloops-per-iteration-bindings.js

```diff
@@ -89,10 +89,10 @@ var tests = [
     {
         name: "const variables in for-in and for-of loops cannot be assigned",
         body: function () {
-            assert.throws(function () { eval("for (const x in { a: 1 }) { x = 1; }"); }, SyntaxError, "assignment to const known at parse time in for-in loop", "Assignment to const");
-            assert.throws(function () { for (const x in { a: 1 }) { eval("x = 1;"); } }, ReferenceError, "assignment to const only known at run time in for-in loop", "Assignment to const");
-            assert.throws(function () { eval("for (const x of [ 0 ]) { x = 1; }"); }, SyntaxError, "assignment to const known at parse time in for-of loop", "Assignment to const");
-            assert.throws(function () { for (const x of [ 0 ]) { eval("x = 1;"); } }, ReferenceError, "assignment to const only known at run time in for-of loop", "Assignment to const");
+            assert.throws(function () { eval("for (const x in { a: 1 }) { x = 1; }"); }, TypeError, "assignment to const known at parse time in for-in loop", "Assignment to const");
```

Comment:
All these should throw runtime TypeErrors now - but these are valid cases and their descriptions are still good.